### ISSUE TITLE
[CI/TEST] Suggestion of a new CI System

### ISSUE THREAD

**Issue by Nexucis** [elastic/elasticsearch-php#693]
Hi,

I would like to propose a migration from travis-ci  to circle-ci. 

I think it would improve the continuous integration of this project and simplify some test or script which had writted.

For exemple with circle-ci we could : 

* start elasticsearch in docker contener and not download it with a bash script.
* replace waiting the running cluster by just a simplify bash script like [this one](https://github.com/Nexucis/es-php-index-helper/blob/master/.circleci/wait_es_up.sh)
     * Separate in multiple job : code analyze, unit test and integration test. 
     * The log would be improved by this job separation. 
     * The seperation in multiple job allow running paralize job too.
     * Running unit test will not depend on a running cluster (maybe it can rejoin the idea put in the issue #625 too)
		 
I discover circle-ci when I need a ci for my index helper, and it really simplify my life when I migrate to this.

You can see an example of a configuration file with integration test and code analyze [here](https://github.com/Nexucis/es-php-index-helper/blob/master/.circleci/config.yml)

If you like this idea, I can put the configuration files in a PR after you allow circle-ci on the repository. 

By the way, I'm not sponsored by circle-ci :)

**Comment by Nexucis** [comment]
ping @polyfractal

**Comment by Nexucis** [comment]
Hi :),

I think that if you are not convinced by this suggestion, perhaps I can make a PR that execute the CI and with that, you can choose what solution you prefer. 

Thoughts ?

**Comment by polyfractal** [comment]
Heya @Nexucis.  I don't think we'll be moving off Travis any time soon.  Not that I have a great love for Travis, but it works and at this point is relatively easy to maintain. 

Re: Docker, we could do similar on Travis but when I tried to get it working last time I tried (many months ago) I ran into issues.  Some with Travis, some with how ES works in docker at the time.  It's fixable I believe... I just didn't have time to do it.

I'm sure CircleCI is nice, but if something's not broken, don't fix it :) 

Thanks though!
